dotCMS/src/main/java/com/dotcms/rest/api/v1/folder/FolderResource.java

```diff
@@ -105,4 +102,25 @@ public final Response createFolders(@Context final HttpServletRequest req,
         return response;
     }
 
+    @GET
+    @Path ("/sitename/{siteName}/uri/{uri : .+}")
+    @JSONP
+    @NoCache
+    @Produces({MediaType.APPLICATION_JSON, "application/javascript"})
+    public final Response loadFolderByURI(@Context final HttpServletRequest req,
+                                          @PathParam("siteName") final String siteName,
+                                          @PathParam("uri") final String uri){
+        Response response = null;
+        final InitDataObject initData = this.webResource.init(null, true, req, true, null);
+        final User user = initData.getUser();
+        try{
+            Folder folder = folderHelper.loadFolderByURI(siteName,user,uri);
+            response = Response.ok( new ResponseEntityView(folder) ).build();
+        } catch (Exception e) { // this is an unknown error, so we report as a 500.
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Catch a list of specific exception subtypes instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2221)